sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -282,6 +285,27 @@ def _test_py_compile_yaml(self, file_base_name):
     finally:
       shutil.rmtree(tmpdir)
 
+  def _test_sample_py_compile_yaml(self, file_base_name):
+    # Jump back to sample dir.
+    test_data_dir = os.path.join(self.core_sample_path, file_base_name)
+    py_file = os.path.join(test_data_dir, file_base_name + '.py')
+    tmpdir = tempfile.mkdtemp()
+    try:
+      target_yaml = os.path.join(tmpdir, file_base_name + '-pipeline.yaml')
+      subprocess.check_call(
+          ['dsl-compile', '--py', py_file, '--output', target_yaml])
+      with open(os.path.join(test_data_dir, file_base_name + '.yaml'),
```

Comment:
The sample folder currently does not host the yaml file, should this be the testdata dir?

Reply:
In this PR it is in the sample folder. Two options of the layout are:
- both .py and .yaml live in sample dir. Pros: When someone updates a .py, it is obvious that corresponding .yaml needs to be updated too.
- .py live in sample dir while .yaml live in testdata/ elsewhere. Pros: cleaner sample dir, all the files under samples/ serve illustrative purpose.

WDYT?

Reply:
I'm leaning towards the second option. The sample directory should be illustrative and clean.
Normally, we do not keep generated code/data in the repo. This particular case serves as the unit tests, which is an exception.
For example, the backend generates swagger codes from the json file and it becomes part of the build process not in the source control repo.

Reply:
Done. PTAL thanks!

> I'm leaning towards the second option. The sample directory should be illustrative and clean.
> Normally, we do not keep generated code/data in the repo. This particular case serves as the unit tests, which is an exception.
> For example, the backend generates swagger codes from the json file and it becomes part of the build process not in the source control repo.

Done. PTAL thx!

Reply:
Thanks